.github/workflows/release.yml

```diff
@@ -0,0 +1,36 @@
+name: Publish Release
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
```

Comment:
Good question. The documentation was a little light on this so I just had to do a test, does look like it is an `or` rather than an `and` so I removed the branches trigger.